src/client/eppo-client.ts

```diff
@@ -651,16 +652,34 @@ export default class EppoClient {
   }
 
   private logBanditAction(banditEvent: IBanditEvent): void {
-    if (!this.banditLogger) {
-      // No bandit logger set; enqueue the event in case a logger is later set
-      if (this.queuedBanditEvents.length < MAX_EVENT_QUEUE_SIZE) {
-        this.queuedBanditEvents.push(banditEvent);
-      }
+    // First we check if this bandit action has been logged before
+    const subjectKey = banditEvent.subject;
+    const flagKey = banditEvent.featureFlag;
+    const banditKey = banditEvent.bandit;
+    const actionKey = banditEvent.action ?? '__eppo_no_action';
+
+    const banditAssignmentCacheProperties = {
+      flagKey,
+      subjectKey,
+      banditKey,
+      actionKey,
+    };
+
+    if (this.banditAssignmentCache?.has(banditAssignmentCacheProperties)) {
+      // Ignore repeat assignment
       return;
     }
-    // If here, we have a logger
+
+    // If here, we have a logger and a new assignment to be logged
     try {
-      this.banditLogger.logBanditAction(banditEvent);
+      if (this.banditLogger) {
+        this.banditLogger.logBanditAction(banditEvent);
+      } else if (this.queuedBanditEvents.length < MAX_EVENT_QUEUE_SIZE) {
+        // If no logger defined, queue up the events (up to a max) to flush if a logger is later defined
+        this.queuedBanditEvents.push(banditEvent);
+      }
+      // Record in the assignment cache, if active, to deduplicate subsequent repeat assignments
+      this.banditAssignmentCache?.set(banditAssignmentCacheProperties);
```

Comment:
@rasendubi no early return before setting the cache entry--instead `if`-`else if` is used to handle the flow control for calling `banditLogger`